### PR TITLE
Allow aliases to qualified imports

### LINKED ISSUE #3669: Alias to qualified module does not work
This program generates an unexpected error:

```py
import os.path
x = os.path
x.isdir('/') # Module has no attribute "isdir"
```

However, this works:

```py
from os import path
x = path
x.isdir('/') # Ok
```

Module aliasing was implemented in #3435.

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Fixes #3669 

This is slightly hacky, since I didn't find anything similar to ``checker.disable_errors()`` in ``semanal`` and I want to avoid duplicate errors like ``x has no attribute y`` and ``name x.y is not defined``.

### CODE REVIEW

File: mypy/semanal.py [hunk 1]
```diff
@@ -3381,15 +3381,16 @@ def tvar_scope_frame(self, frame: TypeVarScope) -> Iterator[None]:
         yield
         self.tvar_scope = old_scope
 
-    def lookup(self, name: str, ctx: Context, suppres_errors: bool = False) -> SymbolTableNode:
+    def lookup(self, name: str, ctx: Context,
+               suppres_errors: bool = False) -> Optional[SymbolTableNode]:
```
Comment (JelleZijlstra) [inline mypy/semanal.py]: suppress, not suppres

**Comment by JukkaL** [comment]
This causes a crash on one of the internal Dropbox codebases. Here's a traceback:

```
Traceback (most recent call last):
  File "/Users/jukka/src/mypy/scripts/mypy", line 6, in <module>
    main(__file__)
  File "/Users/jukka/src/mypy/mypy/main.py", line 50, in main
    res = type_check_only(sources, bin_dir, options)
  File "/Users/jukka/src/mypy/mypy/main.py", line 97, in type_check_only
    options=options)
  File "/Users/jukka/src/mypy/mypy/build.py", line 196, in build
    graph = dispatch(sources, manager)
  File "/Users/jukka/src/mypy/mypy/build.py", line 1769, in dispatch
    process_graph(graph, manager)
  File "/Users/jukka/src/mypy/mypy/build.py", line 2012, in process_graph
    process_stale_scc(graph, scc, manager)
  File "/Users/jukka/src/mypy/mypy/build.py", line 2107, in process_stale_scc
    graph[id].semantic_analysis()
  File "/Users/jukka/src/mypy/mypy/build.py", line 1664, in semantic_analysis
    self.manager.semantic_analyzer.visit_file(self.tree, self.xpath, self.options, patches)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 295, in visit_file
    self.accept(d)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 3641, in accept
    node.accept(self)
  File "/Users/jukka/src/mypy/mypy/nodes.py", line 754, in accept
    return visitor.visit_class_def(self)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 668, in visit_class_def
    defn.defs.accept(self)
  File "/Users/jukka/src/mypy/mypy/nodes.py", line 815, in accept
    return visitor.visit_block(self)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 1522, in visit_block
    self.accept(s)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 3641, in accept
    node.accept(self)
  File "/Users/jukka/src/mypy/mypy/nodes.py", line 859, in accept
    return visitor.visit_assignment_stmt(self)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 1585, in visit_assignment_stmt
    self.process_module_assignment(s.lvalues, s.rvalue, s)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 2577, in process_module_assignment
    rnode = self.lookup_type_node(rval)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 3266, in lookup_type_node
    n = self.lookup_qualified(t.name, expr, suppress_errors=True)
  File "/Users/jukka/src/mypy/mypy/semanal.py", line 3447, in lookup_qualified
    if isinstance(n.node, TypeInfo):
AttributeError: 'NoneType' object has no attribute 'node'
```

I'll try to come up with a repro after the release.

We can go ahead with the release without this PR, but if that happens I'd rather not announce module aliases as a new feature yet.

**Comment by ilevkivskyi** [comment]
@JukkaL Obvious bug from my side, the problem is that ``break`` appeared under ``if not suppress_errors`` while _only_ the error reporting should depend on this. I have fixed this now, you can decide now what to do.

**Comment by JukkaL** [comment]
Thanks! The latest change fixed the crash. If I have to time to review this carefully, this may get included in the release. I'll need to finish a few other tasks first.

File: mypy/semanal.py [hunk 2]
```diff
@@ -3394,23 +3394,25 @@ def tvar_scope_frame(self, frame: TypeVarScope) -> Iterator[None]:
...
+    def lookup(self, name: str, ctx: Context,
+               suppress_errors: bool = False) -> Optional[SymbolTableNode]:
         """Look up an unqualified name in all active namespaces."""
         implicit_name = False
         # 1a. Name declared using 'global x' takes precedence
         if name in self.global_decls[-1]:
             if name in self.globals:
                 return self.globals[name]
-            else:
+            elif not suppress_errors:
```
Comment (gvanrossum) [inline mypy/semanal.py]: I think this ought to be `if` -- it matches the other chunks in this diff and there `if` block ends with `return`.

File: test-data/unit/check-modules.test [hunk 3]
```diff
@@ -1641,6 +1641,40 @@ m = n  # E: Cannot assign multiple modules to name 'm' without explicit 'types.M
 
 [builtins fixtures/module.pyi]
 
+[case testModuleAliasToQualifiedImport]
+import os.path
+
```
Comment (gvanrossum) [inline test-data/unit/check-modules.test]: I recommend way fewer blank lines in test cases.

File: test-data/unit/check-modules.test [hunk 4]
```diff
@@ -1641,6 +1641,40 @@ m = n  # E: Cannot assign multiple modules to name 'm' without explicit 'types.M
 
 [builtins fixtures/module.pyi]
 
+[case testModuleAliasToQualifiedImport]
+import os.path
```
Comment (gvanrossum) [inline test-data/unit/check-modules.test]: I wonder if picking a different package.module name than os.path makes more sense here? I was confused by the whatever() call until I realized this has nothing to do with the stdlib os.path.

**Comment by ilevkivskyi** [comment]
@gvanrossum Thanks! I implemented your comments in a new commit.

**Comment by gvanrossum** [comment]
(Whoops, committed without editing the commit message.)

**Pull request merged**